src/config.js

```diff
@@ -314,42 +314,52 @@ export function newConfig() {
     return Object.assign({}, config);
   }
 
-  /*
-   * Returns the configuration object if called without parameters,
-   * or single configuration property if given a string matching a configuration
-   * property name.  Allows deep access e.g. getConfig('currency.adServerCurrency')
-   *
-   * If called with callback parameter, or a string and a callback parameter,
-   * subscribes to configuration updates. See `subscribe` function for usage.
-   *
-   * The object returned is a deepClone of the `config` property.
-   */
-  function readConfig(...args) {
-    if (args.length <= 1 && typeof args[0] !== 'function') {
-      const option = args[0];
-      const configClone = deepClone(_getConfig());
-      return option ? deepAccess(configClone, option) : configClone;
-    }
-
-    return subscribe(...args);
+  function _getRestrictedConfig() {
+    // This causes reading 'ortb2' to throw an error; with prebid 7, that will almost
+    // always be the incorrect way to access FPD configuration (https://github.com/prebid/Prebid.js/issues/7651)
+    // code that needs the ortb2 config should explicitly use `getAnyConfig`
+    // TODO: this is meant as a temporary tripwire to catch inadvertent use of `getConfig('ortb')` as we transition.
+    // It should be removed once the risk of that happening is low enough.
+    const conf = _getConfig();
+    Object.defineProperty(conf, 'ortb2', {
+      get: function () {
+        throw new Error('invalid access to \'orbt2\' config - use request parameters instead');
```

Comment:
this is applicable to bidders, but what if publishers what to access the config, for example, to achieve #6240

Reply:
the publisher API does not hit this code path - the global `pbjs.getConfig` and `pbjs.readConfig` are aliased to `getAnyConfig` and `readAnyConfig`.